### PR TITLE
Simplify WrappedGaussianProcess

### PR DESCRIPTION
**Pull request by lpereira95**
Besides simplifying some code in `WrappedGaussianProcess`, this PR aims to understand the reason for `TestWrappedGaussianProcess::test_predict_hypersphere` be failing often.

### CODE REVIEW

**Comment by lpereira95** [comment]
My bad here @ninamiolane. This is a very special case, I would say, since we have shapes like `(n_samples_X, n_samples)` and `(n_samples_X, n_targets, n_samples)`. I may disagree with the case `(n_samples, n_targets, n_samples`, but I think it is better to keep consistency.

P.S. @arthurPignet, I need to update my changes in `sample_y` due to this. The simplifications will be smaller.

**Comment by arthurPignet** [comment]
Hello @lpereira95 ! Nice to meet you. 

Yes it seems that there is a confusion on the roleplay of `sample_y`, my documentation is probably unclear/missleading. 

The input shape is (n_samples, n_features), and then for each sample, described by a set of features, you compute the gaussian process associated, and you sample from it potentially several samples. Thus at the end, you have to output (n_samples_X, n_features, n_samples). For *each* set of inputs you sample n_samples.

All the reshapes and repeats and stuff I did can be probably optimised, but the objective of those is to benefit from the vectorization of the Exp. I repeat the basepoint along the first axis to have one basepoint for each sample, in a 2D array. 
 
It is strange that your modification passes the unittests, there are probably not robust enough.

**Comment by lpereira95** [comment]
Thanks @arthurPignet. I will verify if it is possible to optimize the reshapes, but probably not that much. The documentation is good enough, the problem is that I've approached the code with a non-probabilistic mindset, so it was not clear you had samples for each x sample.

I agree the tests should be somehow more robust, as my (bad) changes should have raised errors.

Regarding tests, would it be possible to add an example where `y` only has one feature? Because I believe the code will break given that for `n_features=1` `sklearn`'s `sample_y` outputs `(n_samples_X,  n_samples)`, so the `reshape` you do in line 319 will raise an error.

**Review by arthurPignet — APPROVED** [review]
It is much clearer and simpler! Thank you.

File: geomstats/learning/wrapped_gaussian_process.py [hunk 1]
```diff
@@ -237,49 +237,25 @@ def predict(self, X, return_tangent_std=False, return_tangent_cov=False):
...
-            )
-            y_mean = self.metric.exp(tangent_means, base_point=base_points)
-            result = y_mean
-
-        return result
+        euc_result = self._euclidean_gpr.predict(
+            X, return_cov=return_tangent_cov, return_std=return_tangent_std
+        )
+
+        return_multiple = return_tangent_std + return_tangent_cov
```
[the remark this answers is not in the document]
Comment (lpereira95) [reply geomstats/learning/wrapped_gaussian_process.py:244]: Thanks @arthurPignet!

`or` is more appropriate here (I'll change it). For some reason I was thinking in terms of integers here, which is equivalent, but less clear.

**Pull request merged**